manager/integration/tests/test_scheduling.py

```diff
@@ -765,20 +761,17 @@ def test_data_locality_basic(client, core_api, volume_name, pod, settings_reset)
     wait_for_rebuild_start(client, volume3_name)
     crash_engine_process_with_sigkill(client, core_api, volume3_name)
     volume3 = client.by_id_volume(volume3_name)
-
-    delete_and_wait_pod(core_api, pod3_name)
-    wait_for_volume_detached(client, volume3_name)
-
     err_replica = None
     for replica in volume3.replicas:
         if replica["hostId"] == node1.name:
             err_replica = replica
             break
 
     assert err_replica is not None
-
     wait_for_replica_failed(client, volume3_name, err_replica["name"])
 
+    delete_and_wait_pod(core_api, pod3_name)
+    wait_for_volume_detached(client, volume3_name)
```

Comment:
Make sure to clean up the pod after we done with it. Otherwise, PVC cannot be removed at the in the finalized at the end of the test.